CHANGELOG.md

```diff
@@ -6,12 +6,39 @@ All notable changes to this project will be documented in this file.
 - `cloud-security-xsuaa-integration` library requires JDK 17
 
 ### Breaking Changes
-* Removed packages
-  * `spring-xsuaa-mock` use `java-security-test` instead
+* Removed modules
+  * `spring-xsuaa-mock` &rarr; use [java-security-test](/java-security-test) instead
 * Removed deprecated classes and interfaces
-  * `class` use instead `xxx`
+  * `UserTokenFlow` &rarr; use instead `JwtBearerTokenFlow`
+  * `XSTokenRequest`, `TokenBroker`, `UaaTokenBroker` &rarr; use instead [token-client](/token-client) module to fetch XSUAA tokens via XSUAATokenFlows
+  * `XSUserInfo`, `XSUserInfoAdapter` &rarr; use instead `Token` interface and `Token#getClaimAsString` with `TokenClaims.XSUAA` constants to access XSUAA-specific claims.
+  * `CFEnvironment`, `K8sEnvironment` &rarr; use instead `ServiceBindingEnvironment`
+  * `CFConstants`, `K8sConstants` &rarr; use instead `ServiceConstants`
+  * `TokenBrokerResolver` &rarr; No longer provided. See `spring-security-basic-auth` sample how to write your own implementation. 
+  * `IasXsuaaExchangeBroker` &rarr; Exchange is not supported by XSUAA service anymore.
```

Comment:
Does it support exchanging IAS tokens? Would it work via JwtBearer grant?

Reply:
no, the flow is the other way around, you receive IAS token and then request the Xsuaa to get the access token, the decision logic whether it's a trustworthy id token needs to be implemented by app developers themselves, previously this logic was embedded in xsuaa service which was based on trust configuration from the BTP subaccount. So lets say simplified token xchange flow would look something like this: if my app is bound to xsuaa and identity service I can do ias token validation when the app is accessed with ias token and if its a valid token i can continue with the xsuaa token flows to get the xsuaa token.

Reply:
But which GrantType would be used to fetch the XSUAA access token?

Reply:
depends on the app's requirements it can be CC or Password flow if an user is involved

Reply:
Client credentials of course would be technically possible but only makes sense for system-to-system communication. For authorization inside the resource server itself, it needs the user's scopes which are not available in a client credentials token if I understand correctly.

Reply:
IAS to Xsuaa supported user propagation from IAS to Xsuaa allowing the ias user to get the xsuaa authorization roles, you're right we don't have a drop in replacement for that and it would require decent custom coding to achieve the same. So perhaps better indeed not to mention it 👍🏻